idaes/config.py

```diff
@@ -113,7 +129,31 @@
 }
 """
 
-def new_idaes_config_block():
+cfg = None # the idaes ConfigBlock once it's created by new_idaes_config_block()
+
+
+class ConfigDictJSONEncoder(json.JSONEncoder):
+    """ This class can be used to encode the ConfigDict as json.  It is
+    sufficient for the IDAES config, but not ConfigDicts in general.
+    """
+    def default(self, obj):
+        print(super().default)
+        if isinstance(obj, pyomo.common.config.ConfigDict):
+            d = {}
+            for k, v in obj.items():
+                d[k] = self.default(v)
+            return d
+        elif isinstance(obj, set):
+            return list(obj)
+        return obj
+
+
+def _new_idaes_config_block():
+    """The idaes configuration is stored in a Pyomo ConfigBlock created by this.
```

Comment:
FWIW, you are inconsistent with your use of `ConfigBlock` (the old name) and `ConfigDict` (the new name).  They are the same class, and we should probably just pick one for using everywhere.

Reply:
Yeah.  I noticed that.  Then I started to worry that ConfigBlock would be deprecated and couldn't decide what to do.  Started to think I better use ConfigDict, but didn't really follow through.  I guess I'll go with ConfigBlock for now since that's the way it is in all our models.